src/Cache/Adapter/Sharding.php

```diff
@@ -86,6 +86,19 @@ public function flush(): bool
         return $result;
     }
 
+    /**
+     * @return bool
+     */
+    public function ping(): bool
+    {
+        $result = true;
+        foreach ($this->adapters as $value) {
+            $result = ($value->ping()) ? $result : false;
+        }
+
+        return $result;
```

Comment:
```suggestion
        foreach ($this->adapters as $value) {
            if (!($value->ping())) {
                return false;
            }
        }

        return true;
```

This can stop sooner. If one fails, no need to continue.